backend/controller/dal/dal.go

```diff
@@ -281,14 +267,20 @@ func (t *Tx) Rollback(ctx context.Context) error {
 	return tx.Rollback(ctx)
 }
 
+// GetDB returns the private database interface, which is used by specific, designated
+// packages alongside the dalerrors package to talk to the same database.
+func (d *DAL) GetDB() sql.DBI {
+	return d.db
+}
+
```

Comment:
We should remove this so that people can't bypass the protection the `DAL` provides. Instead we can use the `interface` approach in the comments below

Reply:
done :)